packages/shared/src/UI/components/AssetsManagement/CollectionList.tsx

```diff
@@ -192,17 +197,37 @@ export const CollectionList = memo(function CollectionList({
 
     const currentChainId = chainId ?? defaultChainId ?? (networks.length === 1 ? networks[0].chainId : chainId)
     const { collections, currentCollection, currentCollectionId, setCurrentCollectionId, loading, error, retry } =
-        useCollections(pluginID, currentChainId, account)
+        useCollections({ pluginID, chainId: currentChainId, account, defaultCollectionId })
 
-    const { assetsMapRef, getAssets, getVerifiedBy, loadAssets, loadVerifiedBy } = useUserAssets()
+    const handleCollectionChange = useCallback(
+        (id: string | undefined) => {
+            setCurrentCollectionId(id)
+            onCollectionChange?.(id)
+        },
+        [onCollectionChange],
+    )
+
+    const { assetsMapRef, getAssets, getVerifiedBy, loadAssets, loadVerifiedBy, isAllHidden } = useUserAssets()
 
     const handleInitialRender = useCallback(
         (collection: Web3Helper.NonFungibleCollectionAll) => {
             const id = collection.id!
             // To reduce requests, check if has been initialized
-            if (assetsMapRef.current[id]) return
-            loadAssets(collection)
+            if (assetsMapRef.current[id]?.assets.length) return
             loadVerifiedBy(id)
+            const stateKey = `${id}.${collection.chainId}`
+            const load = async () => {
+                if (assetsMapRef.current[stateKey]?.finished || assetsMapRef.current[stateKey]?.assets.length) return
+                await loadAssets(collection)
+                await delay(30)
+                // The first collectables might have been blocked, if assets of
+                // current collection is empty, try loading another page to fill
+                // the blank.
+                if (!assetsMapRef.current[stateKey].assets.length) {
+                    await load()
+                }
+            }
+            load()
```

Comment:
Maybe add `loadAssetsRecusively` in `useUserAssets()`, too much logic lead to a chunky component.

Reply:
In another coming PR, will check if fetched list are all blocked, if so, load another page.